### ISSUE TITLE
Update email validation

### ISSUE THREAD

**Issue by VBGAMER45** [oneall/social-login-simple-machines-forum#9]
Subs-OneAllSocialLogin.php
From
                // The email address is invalid.
                elseif (!preg_match('~^[0-9A-Za-z=_+\-/][0-9A-Za-z=_\'+\-/\.]*@[\w\-]+(\.[\w\-]+)*(\.[\w]{2,6})$~', $oasl_user_email))

To
                // The email address is invalid.
                elseif (filter_var($oasl_user_email, FILTER_VALIDATE_EMAIL) === false)
                
                
                
                
  This will handle all TLD's better

**Comment by qcybb** [comment]
Agreed - I've modified my local copy, but would like to see this added in the next version.
